core/DataTable/Filter/Truncate.php

```diff
@@ -90,16 +90,18 @@ public function filter($table)
      */
     private function addSummaryRow($table)
     {
-        $table->filter('Sort', array($this->columnToSortByBeforeTruncating, 'desc'));
-
         if ($table->getRowsCount() <= $this->truncateAfter + 1) {
             return;
         }
 
+        $table->filter('Sort', array($this->columnToSortByBeforeTruncating, 'desc', $naturalSort = true, $recursiveSort = false));
```

Comment:
We need to sort the table only if we actually truncate. This saves a lot of time when Archiving or when building Ranges